bitcoin/_posts/0000-00-00-advent-27.md

```diff
@@ -0,0 +1,287 @@
+status quo, so this is certainly one to watch._
+
+## And Then What?
+
+After the release of CTV Soft Fork Client, what goes next?
+
+1. Jeremy takes a vacation for a month.
+1. Sapio continues to improve dramatically, as detailed in the last post.
+1. Work begins on new Opcodes.
+1. More fun applications get built on CTV/Sapio.
+1. Non-Interactive Lightning Channel BOLT drops.
+1. Payment Pool spec drops with LN integration.
+1. (Privacy + Scale)++
+
+## New Opcodes you say?
+
+Yep, new opcodes. As soon as CTV is merged, there are some new features that
+could be tooled into BIPs without much controversy, given their simplicity:
+
+1. New Math Opcodes
+1. OP_AMOUNT
+1. OP_CSFS
+
+These are "universally agreed on" bits of functionality that have very little
+wiggle room in how they might be specified. Therefore their implementation,
+testing, and release is mostly mechanical. This makes them good bets for a
+concrete timeline because they're a rote-development task with few research
+dependencies and easily agreed on design.
+
+With hard work, these could be reviewed and tested in time for Speedy Trial
+signalling in June 2023 (+1 year), with realllllly hard work 6 months earlier
+(but there are conflicting priorities for developer's time -- e.g., building
+stuff with/for CTV and Taproot -- that make that super unlikely).
+
+### What about Anyprevout
+
+Anyprevout is 1000000% not ruled out here for advancing to the consensus stage
+in 2022. There are a couple things happening on Anyprevout that make me suspect
+it might be more primed towards early-mid 2024 best case.
+
+1. Taproot upgraded channels taking some steam away from Eltoo with AJ's
+   proposal to do state compaction via PTLCs.
+1. Disagreement over no justice transactions in the community for high value
+   channels.
+1. Open research on specific Eltoo protocol implementation / need for other
+   tools (like SIGHASH_BUNDLE).
+1. Lack of an open PR and test suite for it.
+1. CSFS + CTV permitting a version of Eltoo.
+
+All of these reasons point to it being highly unlikely APO could be finished on
+the timeline presented for CTV, but also that given a desire to see a working LN
+client (ensuring the protocol is sound end-to-end without modifications) leads
+to additional engineering complexity.
+
+I've heard on the rumour vine some excellent engineers might start working on an
+Eltoo implementation, I believe that work to be of tremendous value to gain
+consensus on anyprevout.
+
+### What about TapLeafUpdateVerify
+
+If I had to estimate, I'd say TLUV needs another year or so of people 'stewing'
+on the possibilities and design before it could move forward. It would then
+probably need at least a year of tinkering to get a well-accepted
+implementation, and then a year for release process. So let's say best case it
+could be a 2024 thing, more likely 2025 given Anyprevout is seen as higher
+priority for the engineering work.
```

Comment:
"... why not see what benefits and functionality we can get from a well-researched, well-tested, and well-reviewed, design specific covenant proposal like OP_CTV which also gives us congestion control and block space efficiencies in the meantime" or something like that :P

Reply:
maybe 'given that this seems like the best case for APO, if we have the ability to deliver CTV now, ...

Reply:
I think this comment was on TLUV, but I felt the response nests better in APO.